python/hail/dataset.py

```diff
@@ -3419,6 +3419,98 @@ def storage_level(self):
         return self._jvds.storageLevel()
 
     @handle_py4j
+    def set_va_attribute(self, ann_path, key, value):
+        """Sets an attribute for a variant annotation.
+        Attributes are key/value pairs that can be attached to a variant annotation field.
+
+        The following attributes are read from the VCF header when importing a VCF and written
+        to the VCF header when exporting a VCF:
+        - INFO fields attributes (attached to (`va.info.*`)):
+            - 'Number': The arity of the field. Can take values
+            `0` (Boolean flag),
+            `1` (single value),
+            `R` (one value per allele, including the reference),
+            `A` (one value per non-reference allele),
+            `G` (one value per genotype), and
+            `.` (any number of values)
+                - When importing: The value in read from the VCF INFO field definition
+                - When exporting: The default value is `0` for **Boolean**, `.` for **Arrays** and 1 for all other types
+            - 'Description' (default is '')
+        - FILTER entries in the VCF header are generated based on the attributes of `va.filters`. Each key/value pair in the attributes will generate a FILTER entry in the VCF with ID = key and Description = value.
+
+        **Examples**
+
+        Consider the following command which a filter and an annotation to the VDS (we're assuming a split VDS for simplicity):
+        1) an INFO field `AC_HC`, which stores the allele count of high confidence genotypes (DP >= 10, GQ >= 20) for each non-reference allele,
+        2) a filter `HardFilter` that filters all sites with the [GATK suggested hard filters](http://gatkforums.broadinstitute.org/gatk/discussion/2806/howto-apply-hard-filters-to-a-call-set):
+            - For SNVs: QD < 2.0 || FS < 60 || MQ < 40 || MQRankSum < -12.5 || ReadPosRankSum < -8.0
+            - For Indels (and other complex): QD < 2.0 || FS < 200.0 || ReadPosRankSum < 20.0
+        >>> vds = hc.read('data/example.vds')
```

Comment:
I don't think you're allowed to overwrite the `vds` namespace of the test vds. `vds` already exists as this.